User documentation for GPU integration with Pyccel (#1290)
pyccel/pyccel: developer_docs/GPU_getting_started.md

# Getting started GPU

## Decorators

  * Kernel

    In Pyccel, the `@kernel` decorator is used to indicate that a function should be treated as a CUDA kernel. A CUDA kernel is a function that is executed on a GPU and is typically used to perform a parallel computation over a large dataset.

>     The `@kernel` decorator is used to indicate to Pyccel that the function should be compiled as a CUDA kernel, and that it can be launched on a GPU using the appropriate syntax. For example, if we have a function decorated with `@kernel` like this:

Reviewer: There seems to be some repetition. Could we rephrase it to something like this, for example ?
```suggestion
By applying the @kernel decorator to a function, Pyccel recognizes it as a CUDA kernel that can be launched on a GPU using the appropriate syntax.
```

  * cuda.blockDim(dim): Returns the size of the thread block in a given dimension dim. dim is an integer between 0 and the number of dimensions of the thread block minus one.

  * cuda.gridDim(dim): Returns the size of the grid in a given dimension dim. dim is an integer between 0 and the number of dimensions of the grid minus one.

> These built-in variables are provided by the CUDA runtime and are available to all CUDA kernels, regardless of the programming language being used. They are used to obtain information about the execution configuration of the kernel, such as the indices and dimensions of the threads and blocks. They are an important part of the CUDA programming model and are often used to compute global indices and other information about the execution configuration of the kernel.

```Python
from pyccel import cuda
from pyccel.decorators import kernel

@kernel
def my_kernel(x: 'float64[:]', y: 'float64[:]', out: 'float64[:]'):
    i = cuda.threadIdx(0) + cuda.blockIdx(0) * cuda.blockDim(0)
    if i >= x.shape[0]:
        return
    out[i] = x[i] + y[i]
```

> The kernel uses the `cuda.threadIdx(0)`, `cuda.blockIdx(0)`, and `cuda.blockDim(0)` built-in variables to compute the global index of the current thread within the input arrays. 

Author: yeah i think this should be added but in a way that doesn't suggest that using `cuda.threadIdx(dim)` where `dim=0` is related to the array size.

> The kernel uses the `cuda.threadIdx(0)`, `cuda.blockIdx(0)`, and `cuda.blockDim(0)` built-in variables to compute the global index of the current thread within the input arrays. 

Author: If it is clear that that paragraph only describe the code above your change would make sense but it shouldn't mean that it is a rule on how to use the built-in variables.
